tests/netpyne-tests.js

```diff
@@ -1,505 +1,2565 @@
+    populateRangeComponent(test, "CellParams") // populate RangeComponent
+  })
+}
+
+//----------------------------------------------------------------------------//
+function checkCellParamsValues(test, name, cellType, cellModel, pop, rangeEmpty = false) {
+  casper.then(function() {
+    getInputValue(test, "cellRuleName", name)
+    getSelectFieldValue(test, "netParams.cellParams[\'" + name + "\'][\'conds\'][\'cellType\']", cellType)
+    getSelectFieldValue(test, "netParams.cellParams[\'" + name + "\'][\'conds\'][\'cellModel\']", cellModel)
+    getSelectFieldValue(test, "netParams.cellParams[\'" + name + "\'][\'conds\'][\'pop\']", pop)
+  })
+  casper.then(function() {
+    if (rangeEmpty) {
+      checkRangeComponentIsEmpty(test, "CellParams")
+    } else {
+      testRangeComponent(test, "CellParams")
+    }
+  })
+}
+
+//----------- going to section page ----------
+function testSectionAndMechanisms(test) {
+  message("going to section page")
+  casper.thenClick('button[id="cellParamsGoSectionButton"]', function() { //go to "section" page
+    test.assertExist('button[id="newCellRuleSectionButton"]', "landed in section page")
+  })
+  casper.thenClick('#newCellRuleSectionButton', function() { //create section 1
+    this.echo("creating 2 sections")
+    getInputValue(test, "cellParamsSectionName", "Section")
+  });
+  casper.thenClick('#newCellRuleSectionButton', function() { //create section 2
+    getInputValue(test, "cellParamsSectionName", "Section 2")
+  });
+  casper.thenClick('button[id="Section"]') //focus on section 1
+
+  //----------- going to "Geometry" tab in "section" page ----------
+  casper.then(function() {
+    active.buttonID = "newCellRuleSectionButton"
+    active.tabID = "sectionGeomTab"
+  })
+
+  casper.thenClick("#sectionGeomTab", function() { //go to "geometry" tab in "section" page
+    this.echo("going to Geometry tab")
+  })
+  casper.then(function() { // polulate geometry
+    populateSectionGeomTab(test)
+  })
+
+  casper.then(function() { // go to general tab and come back to geometry tab
+    leaveReEnterTab(test, "sectionGeomTab", "netParams.cellParams[\'CellRule\'][\'secs\'][\'Section\'][\'geom\'][\'diam\']", "sectionGeneralTab", "cellParamsSectionName")
+  })
+
+  casper.then(function() { // check values remain the same
+    checkSectionGeomTabValues(test, "CellRule", "Section")
+  })
+  casper.then(function() { // try to delete an item from pt3d component
+    deleteListItem(test, "netParams.cellParams[\'CellRule\'][\'secs\'][\'Section\'][\'geom\'][\'pt3d\']1")
+  })
+
+  casper.thenClick('button[id="Section 2"]', function() { // change to rule 2
+    this.echo("go to section 2 -> values must be empty")
+    this.wait(2500) // let python populate fields  
+  })
+
+  casper.then(function() { // check values must be empty
+    checkSectionGeomTabValues(test, "CellRule", "Section 2", "", "", "", "", "", "")
+  })
+
+  casper.thenClick('button[id="Section"]', function() { // back to s
```

Comment:
This is more like renameEntity or sth like that, right?

Reply:
yeah, but I created a function to rename rules like popParams rules, conParams rules, but because it also works for renaming sections and for renaming mechanisms, I just used it there